pandas/core/arrays/period.py

```diff
@@ -642,12 +647,19 @@ def astype(self, dtype, copy: bool = True):
             return self.asfreq(dtype.freq)
         return super().astype(dtype, copy=copy)
 
-    def searchsorted(self, value, side="left", sorter=None) -> np.ndarray:
-        value = self._validate_searchsorted_value(value).view("M8[ns]")
+    def searchsorted(
+        self,
+        value: ArrayLike | object,
```

Comment:
i think de-facto the scalar needs to be either Period or Period-castable

Reply:
Yes, but the `ExtensionArray` type has to allow any `object`, so the subtype does as well, and then it should raise if someone passes an incompatible object type.

Reply:
> has to allow any object

aren't we supposed to use `Any` for that?

> so the subtype does as well, and then it should raise if someone passes an incompatible object type

so if a subclass is more restrictive in what it accepts (without raising), we can't have that be reflected in the annotation?

Reply:
> > has to allow any object
> 
> aren't we supposed to use `Any` for that?

If we use `Any`, then that tells the type checker to ignore type checking on the result.  If we use `object`, then it will continue type checking.
> 
> > so the subtype does as well, and then it should raise if someone passes an incompatible object type
> 
> so if a subclass is more restrictive in what it accepts (without raising), we can't have that be reflected in the annotation?

Yes and no.  The full signature has to accept the same type as the parent, and can widen that argument type, but not narrow it.  But, I think we can use an `overload` signature to indicate that if a specific type is passed, it creates a result of that type.  So the  master signature could accept `object`, and then you can have an `overload` to correspond to the narrower type.